sdk/monitor/azure-monitor-opentelemetry-exporter/tests/statsbeat/test_statsbeat.py

```diff
@@ -73,50 +82,62 @@ def test_collect_statsbeat_metrics_exists(self):
-            mr = mp._sdk_config.metric_readers[0]
-            stats_exporter = mr._exporter
-            self.assertEqual(
-                stats_exporter._instrumentation_key,
-                _statsbeat._DEFAULT_NON_EU_STATS_CONNECTION_STRING.split(";")[0].split("=")[1]
-            )
-            self.assertEqual(
-                stats_exporter._endpoint,
-                _statsbeat._DEFAULT_NON_EU_STATS_CONNECTION_STRING.split(";")[1].split("=")[1]   # noqa: E501
-            )
+        mp = _statsbeat._STATSBEAT_METER_PROVIDER
+        mr = mp._sdk_config.metric_readers[0]
+        stats_exporter = mr._exporter
+        self.assertEqual(
+            stats_exporter._instrumentation_key,
+            _statsbeat._DEFAULT_NON_EU_STATS_CONNECTION_STRING.split(";")[0].split("=")[1]
+        )
+        self.assertEqual(
+            stats_exporter._endpoint,
+            _statsbeat._DEFAULT_NON_EU_STATS_CONNECTION_STRING.split(";")[1].split("=")[1]   # noqa: E501
+        )
 
     def test_collect_statsbeat_metrics_eu(self):
         exporter = mock.Mock()
         exporter._instrumentation_key = "1aa11111-bbbb-1ccc-8ddd-eeeeffff3333"
         exporter._endpoint = "https://northeurope-0.in.applicationinsights.azure.com/"
         self.assertIsNone(_statsbeat._STATSBEAT_METER_PROVIDER)
         with mock.patch.dict(
-                os.environ, {
-                    "APPLICATION_INSIGHTS_STATS_CONNECTION_STRING": "",
-                }):
+            os.environ, {
+                "APPLICATION_INSIGHTS_STATS_CONNECTION_STRING": "",
+            }), \
+            mock.patch.object(_StatsbeatMetrics, 'init_non_initial_metrics'), \
+            mock.patch.object(_StatsbeatMetrics, '_get_attach_metric'):
             _statsbeat.collect_statsbeat_metrics(exporter)
-            mp = _statsbeat._STATSBEAT_METER_PROVIDER
-            mr = mp._sdk_config.metric_readers[0]
-            stats_exporter = mr._exporter
-            self.assertEqual(
-                stats_exporter._instrumentation_key,
-                _statsbeat._DEFAULT_EU_STATS_CONNECTION_STRING.split(";")[0].split("=")[1]
-            )
-            self.assertEqual(
-                stats_exporter._endpoint,
-                _statsbeat._DEFAULT_EU_STATS_CONNECTION_STRING.split(";")[1].split("=")[1]   # noqa: E501
-            )
+        mp = _statsbeat._STATSBEAT_METER_PROVIDER
+        mr = mp._sdk_config.metric_readers[0]
+        stats_exporter = mr._exporter
+        self.assertEqual(
+            stats_exporter._instrumentation_key,
+            _statsbeat._DEFAULT_EU_STATS_CONNECTION_STRING.split(";")[0].split("=")[1]
+        )
+        self.assertEqual(
+            stats_exporter._endpoint,
+            _statsbeat._DEFAULT_EU_STATS_CONNECTION_STRING.split(";")[1].split("=")[1]   # noqa: E501
+        )
 
+_StatsbeatMetrics_COMMON_ATTRS = dict(
+    _StatsbeatMetrics._COMMON_ATTRIBUTES
+)
+_StatsbeatMetrics_NETWORK_ATTRS = dict(
+    _StatsbeatMetrics._NETWORK_ATTRIBUTES
+)
```

Comment:
If this is only needed for TestStatsbeatMetrics but not TestStatsbeat, is there a reason we don't put it in the TestStatsbeatMetrics  setup?

Reply:
`setUp` is run for every single test. We created those two variables to hold the ORIGINAL values of `_COMMON_ATTRIBUTES` and `_NETWORK_ATTRIBUTES` so we can reset them every time a test is run.

Reply:
Got it. I feel like it would make sense to put those values inside the TestStatsbeatMetrics class before setUp.

Reply:
Yeah we could. Just chose not too because accessing `TestStatsbeatMetrics` static method is kind of weird. Not a huge opinion though.